lib/handler.js

```diff
@@ -73,52 +73,61 @@ class Handler {
     // let the user know that we are validating if PR is mergeable
-
-    var config = await Configuration.instanceWithContext(context)
-
-    let settings = config.settings.mergeable.pull_requests || config.settings.mergeable
-    let validators = []
-    let excludes = (settings.exclude)
-      ? settings.exclude.split(',').map(val => val.trim()) : []
-    let includes = [ 'approvals', 'title', 'label', 'milestone', 'description', 'projects', 'assignee', 'dependent' ]
-      .filter(validator => excludes.indexOf(validator) === -1)
-
-    includes.forEach(validator => {
-      validators.push(require(`../lib/${validator}`)(pullRequest, context, settings))
-    })
-
-    console.info(config)
-    return Promise.all(validators).then(results => {
-      let failures = results.filter(validated => !validated.mergeable)
-
-      let status, description
-      if (failures.length === 0) {
-        status = 'success'
-        description = 'Okay to merge.'
-      } else {
-        status = 'failure'
-        description = `## Mergeable has found the following failed checks`
-        for (let validated of failures) {
-          description += `\n - ${validated.description}`
+    try {
+      var config = await Configuration.instanceWithContext(context)
+      let settings = config.settings.mergeable.pull_requests || config.settings.mergeable
+      let validators = []
+      let excludes = (settings.exclude)
+        ? settings.exclude.split(',').map(val => val.trim()) : []
+      let includes = [ 'approvals', 'title', 'label', 'milestone', 'description', 'projects', 'assignee', 'dependent' ]
+        .filter(validator => excludes.indexOf(validator) === -1)
+
+      includes.forEach(validator => {
+        validators.push(require(`../lib/${validator}`)(pullRequest, context, settings))
+      })
+      console.info(config)
+      return Promise.all(validators).then(results => {
+        let failures = results.filter(validated => !validated.mergeable)
+
+        let status, description
+        if (failures.length === 0) {
+          status = 'success'
+          description = 'Okay to merge.'
+        } else {
+          status = 'failure'
+          description = `## Mergeable has found the following failed checks`
+          for (let validated of failures) {
+            description += `\n - ${validated.description}`
+          }
+          description += `\n ___ \n **Please address the problems found above!**`
         }
-        description += `\n ___ \n **Please address the problems found above!**`
-      }
 
+        checks.update(
+          context,
+          checkRunResult.data.id,
+          'Mergeable',
+          'completed',
+          status,
+          {
+            title: `Result: ${status}`,
+            summary: description})
+
+        console.info({state: status, description: description})
+      }).catch(error => {
+        // (jusx) This should never ever happen. Log it.
+        console.error(error)
```

Comment:
There should probably be a `checks.update` here for general errors. Though likely no action required because it isn't really configuration related. This will likely be like a "HTTP 500".

Reply:
removed that `catch` the try and catch block should catch this error as well